21-game-project/contract/src/contract.rs

```diff
@@ -35,79 +41,123 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
     match msg {
-        ExecuteMsg::Deck {} => build_deck(deps, env),
+        ExecuteMsg::Deck {} => build_deck(deps, env, _info),
+        ExecuteMsg::Increment {} => increment_index(deps, _info),
+        ExecuteMsg::Enough {} => enough_cards(deps, _info),
     }
 }
 
-pub fn build_deck(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
-        let address = SECRET_ADDRESS
-        .add_suffix("secret1grgcderf32vn8zpcy3fu3k96cxaffynrncyjf6".as_bytes());
+pub fn build_deck(deps: DepsMut, env: Env, _info: MessageInfo) -> Result<Response, ContractError> {
         let mut cards: Vec<u8> = vec![0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 5, 5, 5, 5, 6, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8, 8];
         let rnd_vec = env.block.random.unwrap().0;
         let mut tempdeck: Vec<u8> = Vec::new();
+
+        tempdeck.push(0); //is_game_finished
+        tempdeck.push(6); //index 
+        tempdeck.push(0); //card_sum  
+        tempdeck.push(0); //card_sum_with_ace
```

Comment:
I would call it state, not deck. Also it's not temporary, we push it into the storage.